influxdb3_server/src/lib.rs

```diff
@@ -319,15 +319,191 @@ mod tests {
         shutdown.cancel();
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn write_lp_tests() {
```

Comment:
I think the cases covered in this test are solid, but I have two concerns:
1. The assertions on error responses may be a bit brittle
2. There is a lot of work in the setup for this test

I think we may be well served to move this into an end-to-end (e2e) test where we can run the binary and leverage defaults to do a lot of the setup work for us.

I have done some work on building out some re-usable test helpers to streamline the e2e tests in #24678. If we like that, perhaps this test can be ported to an e2e test in a follow on issue?

Reply:
I think porting that over in an e2e test later would be preferable yeah. I opened up a ticket for it https://github.com/influxdata/influxdb/issues/24694